### PR TITLE
Enabling dynamic queryparam addition to anchour links 

### PR DESCRIPTION
**Pull request by rajkumarsrk**
- Providing option to add dynamic query parameters for source page attribution .This is done by enabling `data-amp-params` to get added with the anchour links href

https://github.com/ampproject/amphtml/issues/9673

### CODE REVIEW

File: src/service/url-replacements-impl.js [hunk 1]
```diff
@@ -785,9 +785,10 @@ export class UrlReplacements {
       'CLIENT_ID': true,
       'QUERY_PARAM': true,
     };
+    const additionalURLParameters = element.getAttribute('data-amp-addparams');
```
Comment (jridgewell) [inline src/service/url-replacements-impl.js]: Should have caught this last time. Please rename to `additionalUrlParameters` (the `Url` part).

File: src/service/url-replacements-impl.js [hunk 2]
```diff
@@ -785,9 +785,10 @@ export class UrlReplacements {
...
     const whitelist = this.getWhitelistForElement_(
         element, supportedReplacements);
-    if (!whitelist) {
+    if (!whitelist && !additionalURLParameters) {
```
Comment (jridgewell) [inline src/service/url-replacements-impl.js]: So the paths below aren't necessary (or not always necessary) if we only have `aditionalURLParameters`.

- We don't need to build the href again if we've already done it
- We don't need a secure URL (no privacy issues)
- We don't need to expand the URL

We really need to move the `addParamsToUrl` stuff up, then we can do a few things:

```js
if (!whitelist && !additional) {
  return;
}
let herf = assertString(element[original_href_property] || getAttribute('href'));
if (!element[original_href_property]) {
  element[original_href_property) = href;
}
if (additional) {
  href = addParams(additional);
}
if (whitelist) {
  assertSecure();
  assertAllowedOrigin();
  href = expandSync();
}
element.href = href;

Comment (rajkumarsrk) [inline src/service/url-replacements-impl.js]: PR has been updated with these changes

**Pull request merged**